### PR TITLE
part of cam6_3_065: more GW diags

### PR DESCRIPTION
**Pull request by JulioTBacmeister**
Added additional diagnostics to orographic GW parameterization

### CODE REVIEW

File: src/physics/cam/gw_drag.F90 [hunk 1]
```diff
@@ -658,14 +658,58 @@ subroutine gw_init()
...
+     call addfld ('GBXAR1',     horiz_only,  'I','km+2', &
+          'grid box area')
+
+     call addfld ('Fr1_DIAG',     horiz_only,  'I','1', &
+          'grid box area')
+     call addfld ('Fr2_DIAG',     horiz_only,  'I','1', &
+          'grid box area')
+     call addfld ('Frx_DIAG',     horiz_only,  'I','1', &
+          'grid box area')
+
```
Comment (cacraigucar) [inline src/physics/cam/gw_drag.F90:698]: These all have the same long name of `grid box area`.  Is this correct?  I suspect the last three need a an updated long name.

File: src/physics/cam/gw_drag.F90 [hunk 2]
```diff
@@ -658,14 +658,58 @@ subroutine gw_init()
...
      call addfld('VEGW',  (/ 'lev' /) , 'A'  ,'1/s' ,  &
           'Merdional wind profile-entry to GW ' )
      call register_vector_field('UEGW','VEGW')
      call addfld('TEGW',  (/ 'lev' /) , 'A'  ,'K' ,  &
           'Temperature profile-entry to GW ' )
-
+     call addfld('ZEGW',  (/ 'ilev' /) , 'A'  ,'m' ,  &
+          'interface geopotential heights in GW code ' )
+     call addfld('ZMGW',  (/ 'ilev' /) , 'A'  ,'m' ,  &
+          'interface geopotential heights in GW code ' )
```
Comment (cacraigucar) [inline src/physics/cam/gw_drag.F90:709]: There is a sublte name difference on these.  The long name should reflect the difference.

File: src/physics/cam/gw_drag.F90 [hunk 3]
```diff
@@ -2158,6 +2208,30 @@ subroutine gw_rdg_calc( &
...
+         ! Ridge quantities - don't change.  Written for convenience
+         call outfld('MXDIS1' , mxdis(:,nn) ,  ncol, lchnk)
+         call outfld('ANGLL1' , angll(:,nn) ,  ncol, lchnk)
+         call outfld('ANIXY1' , anixy(:,nn) ,  ncol, lchnk)
+         call outfld('HWDTH1' , hwdth(:,nn) ,  ncol, lchnk)
+         call outfld('CLNGT1' , clngt(:,nn) ,  ncol, lchnk)
+         call outfld('GBXAR1' , gbxar ,        ncol, lchnk)
+      endif
+
+      if (nn == 1) then
```
Comment (cacraigucar) [inline src/physics/cam/gw_drag.F90:2233]: Why not have just one `if (nn==1) then` block?  I there should only be one if block here.

File: src/physics/cam/gw_rdg.F90 [hunk 4]
```diff
@@ -564,9 +564,11 @@ subroutine gw_rdg_belowpeak(ncol, band, rdg_cd_llb, &
   ! Wave Reynolds stress.
   real(r8), intent(inout) :: tau(ncol,-band%ngwv:band%ngwv,pver+1)
   ! Top of low-level flow layer.
-  real(r8), intent(inout) :: tlb(ncol)
+  real(r8), intent(in)    :: tlb(ncol)
+     !real(r8), intent(inout) :: tlb(ncol)
```
Comment (cacraigucar) [inline src/physics/cam/gw_rdg.F90:568]: Remove obsolete code

File: src/physics/cam/gw_rdg.F90 [hunk 5]
```diff
@@ -564,9 +564,11 @@ subroutine gw_rdg_belowpeak(ncol, band, rdg_cd_llb, &
...
   ! Bottom of linear wave region.
-  real(r8), intent(inout) :: bwv(ncol)
+  real(r8), intent(in)    :: bwv(ncol)
+     !real(r8), intent(inout) :: bwv(ncol)
```
Comment (cacraigucar) [inline src/physics/cam/gw_rdg.F90:571]: Remove obsolete code

File: src/physics/cam/gw_drag.F90 (hunk 1 above)
Comment (JulioTBacmeister) [reply src/physics/cam/gw_drag.F90:698]: Yes long names for these should be
Fr1_diag: "Froude number defining linear wave amplitude"
Fr2_diag: "Froude number defining blocked flow layer"
Frx_diag: "Obstacle Froude number"

File: src/physics/cam/gw_drag.F90 (hunk 2 above)
Comment (JulioTBacmeister) [reply src/physics/cam/gw_drag.F90:709]: ZMGW is actually incorrectly specified here.  This should actually be:

 call addfld('ZMGW',  (/ 'lev' /) , 'A'  ,'m' ,  &
          'mid-layer geopotential heights in GW code ' )

File: src/physics/cam/gw_drag.F90 (hunk 3 above)
Comment (JulioTBacmeister) [reply src/physics/cam/gw_drag.F90:2233]: Sure.

File: src/physics/cam/gw_rdg.F90 (hunk 4 above)
Comment (JulioTBacmeister) [reply src/physics/cam/gw_rdg.F90:568]: yes

File: src/physics/cam/gw_rdg.F90 (hunk 5 above)
Comment (JulioTBacmeister) [reply src/physics/cam/gw_rdg.F90:571]: yes

**Review by peverwhee — APPROVED** [review]

**Review by gold2718 — APPROVED** [review]

**Review by nusbaume — APPROVED** [review]

**Review by cacraigucar — APPROVED** [review]

**Pull request merged**